dvc/repo/experiments/executor/ssh.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import os
+import posixpath
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Callable, Optional
+
+# from funcy import cached_property, first
+from funcy import first
+
+from dvc.fs.ssh import SSHFileSystem
+from dvc.repo.experiments.base import (
+    EXEC_BRANCH,
+    EXEC_CHECKPOINT,
+    EXEC_HEAD,
+    EXEC_MERGE,
+    EXEC_NAMESPACE,
+)
+
+from .base import BaseExecutor
+
+if TYPE_CHECKING:
+    # from multiprocessing import Queue
+
+    from dvc.machine import MachineManager
+    from dvc.scm.git import Git
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def _sshfs(fs_factory, **kwargs):
+    if fs_factory:
+        with fs_factory() as fs:
+            yield fs
+        return
+    yield SSHFileSystem(**kwargs)
+
+
+class SSHExecutor(BaseExecutor):
+    """SSH experiment executor."""
+
+    WARN_UNTRACKED = True
+    QUIET = True
+
+    def __init__(
+        self,
+        *args,
+        host: Optional[str] = None,
+        port: Optional[int] = None,
+        username: Optional[str] = None,
+        fs_factory: Optional[Callable] = None,
+        **kwargs,
+    ):
+        assert host
+
+        self.host: str = host
+        self.port = port
+        self.username = username
+        self._fs_factory = fs_factory
+
+        super().__init__(*args, **kwargs)
+        logger.debug("Init SSH executor for host '%s'", self.host)
+
+    @classmethod
+    def from_machine(
+        cls,
+        manager: "MachineManager",
+        machine_name: Optional[str],
+        *args,
+        **kwargs,
+    ):
+        from shortuuid import uuid
+
+        name = kwargs.get("name") or "dvc-exp"
+        kwargs["root_dir"] = f"{name}-executor-{uuid()}"
+        kwargs.update(manager.get_executor_kwargs(machine_name))
+        return cls(*args, **kwargs)
+
+    def sshfs(self):
+        return _sshfs(self._fs_factory, host=self.host, port=self.port)
+
+    @property
+    def git_url(self) -> str:
+        user = f"{self.username}@" if self.username else ""
+        port = f":{self.port}" if self.port is not None else ""
+        path = f"{self.root_dir}" if self.root_dir else ""
+        if path and not posixpath.isabs(path):
+            path = f"/~/{path}"
+        return f"ssh://{user}{self.host}{port}{path}"
+
+    @staticmethod
+    def _git_client_args(fs):
+        kwargs = {
+            "password": fs.fs_args.get("password"),
+            "key_filename": first(fs.fs_args.get("client_keys", [])),
+        }
+        if os.name != "win32":
+            kwargs["ssh_command"] = (
+                "ssh -o UserKnownHostsFile=/dev/null "
+                "-o StrictHostKeyChecking=no"
+            )
```

Comment:
TODO: this needs to be moved into dvc-machine specific ssh args before this PR is mergeable (we don't want to disable host checking for non-terraform generated hosts)

Reply:
Upon further review, this probably does need to be handled either here in the ssh executor or in the dulwich git backend. Dulwich chooses an ssh vendor type (openssh client, putty, etc) depending on local platform, so these args are really per git/executor and not per dvc-machine